ext/pdo/pdo_stmt.c

```diff
@@ -1643,6 +1571,24 @@ PHP_METHOD(PDOStatement, getColumnMeta)
 }
 /* }}} */
 
+void pdo_stmt_free_default_fetch_mode(pdo_stmt_t *stmt)
+{
+	enum pdo_fetch_type default_fetch_mode = stmt->default_fetch_type & ~PDO_FETCH_FLAGS;
+	if (default_fetch_mode == PDO_FETCH_INTO) {
+		/* This can happen if the fetch flags are set via PDO::setAttribute()
+		 * $pdo->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_INTO);
+		 * See ext/pdo/tests/bug_38253.phpt */
+		if (EXPECTED(stmt->fetch.into != NULL)) {
+			OBJ_RELEASE(stmt->fetch.into);
+		}
+	} else if (default_fetch_mode == PDO_FETCH_CLASS) {
+		if (stmt->fetch.cls.ctor_args != NULL) {
+			zend_hash_release(stmt->fetch.cls.ctor_args);
```

Comment:
This can be:

```suggestion
			zend_array_release(stmt->fetch.cls.ctor_args);
```

as they're real arrays, not storing any persistent or non-zval stuff

Reply:
Also: can ctor_args be cyclic? If so, then this is not enough